repos/system_upgrade/el7toel8/actors/sourcebootloaderscanner/libraries/sourcebootloaderscanner.py

```diff
@@ -26,20 +50,28 @@ def scan_boot_entries():
         )
 
     boot_entries = []
-    # Identify the available boot entries by searching for their titles in the grubby output
+    boot_entry_data = None
     for output_line in grubby_output['stdout']:
-        # For now it is sufficient to look only for the titles as that is the only
-        # information we use. If need be, we would have to parse the structure
-        # of the grubby output into sections according to the `index` lines
-        if output_line.startswith('title='):
-            boot_entry = output_line[6:]  # Remove the `title=` prefix
-
-            # On s390x grubby produces quotes only when needed (whitespace in
-            # the configuration values), on x86 the values are quoted either way
-            boot_entry = boot_entry.strip('\'"')
+        if output_line == 'non linux entry':
+            # Grubby does not display info about non-linux entries
+            # Such an entry is not problematic from our PoV, therefore, skip it
+            boot_entry_data = {}
+            continue
 
-            boot_entries.append(BootEntry(title=boot_entry))
+        prop, value = parse_grubby_output_line(output_line)
+        if prop == 'index':
+            # Start of a new boot entry section
+            if boot_entry_data is not None:
+                boot_entries.append(
+                    BootEntry(title=boot_entry_data.get('title', ''),  # In theory, the title property can be missing
+                              kernel_image=boot_entry_data['kernel']))
```

Comment:
Hmm correct me if I am wrong please - given the way code is written now we can have KeyError at L67 when `boot_entry_data={}` and we are trying to retrieve `kernel`.

Theoretically, if we have been getting only 'non linux entry' in grubby_output['stdout'], then the first run with `prop == 'index'` at L62 will have `boot_entry_data={}` and that's when we are running into trouble with KeyError.

Reply:
Brilliant observation, thanks! I'll fix it right away. 

I remember that I've retrofitted this the 'non linux entry' only after I've realized that testing farm VMs have such entries, so that might be why I did not realize this might be a problem.